pyiron_atomistics/vasp/base.py

```diff
@@ -400,81 +392,27 @@ def write_input(self):
-        """
-        Collects the outputs and stores them to the hdf file
-        """
-        if self.structure is None or len(self.structure) == 0:
-            try:
-                self.structure = self.get_final_structure_from_file(
-                    cwd=cwd, filename="CONTCAR"
-                )
-            except IOError:
-                self.structure = self.get_final_structure_from_file(
-                    cwd=cwd, filename="POSCAR"
-                )
-            self._sorted_indices = np.array(range(len(self.structure)))
-        self._output_parser.structure = self.structure.copy()
-        try:
-            self._output_parser.collect(
-                directory=cwd, sorted_indices=self.sorted_indices
-            )
-        except VaspCollectError:
-            self.status.aborted = True
-            raise
-        # Try getting high precision positions from CONTCAR
-        try:
-            self._output_parser.structure = self.get_final_structure_from_file(
-                cwd=cwd,
-                filename="CONTCAR",
-            )
-        except (IOError, ValueError, FileNotFoundError):
-            pass
-
-        # Bader analysis
-        if os.path.isfile(os.path.join(cwd, "AECCAR0")) and os.path.isfile(
-            os.path.join(cwd, "AECCAR2")
-        ):
-            bader = Bader(
-                working_directory=self.working_directory, structure=self.structure
-            )
-            try:
-                charges_orig, volumes_orig = bader.compute_bader_charges()
-            except ValueError:
-                warnings.warn("Invoking Bader charge analysis failed")
-                self.logger.warning("Invoking Bader charge analysis failed")
-            else:
-                charges, volumes = charges_orig.copy(), volumes_orig.copy()
-                charges[self.sorted_indices] = charges_orig
-                volumes[self.sorted_indices] = volumes_orig
-                if (
-                    "valence_charges"
-                    in self._output_parser.generic_output.dft_log_dict.keys()
-                ):
-                    valence_charges = self._output_parser.generic_output.dft_log_dict[
-                        "valence_charges"
-                    ]
-                    # Positive values indicate electron depletion
-                    self._output_parser.generic_output.dft_log_dict["bader_charges"] = (
-                        valence_charges - charges
-                    )
-                self._output_parser.generic_output.dft_log_dict["bader_volumes"] = (
-                    volumes
-                )
-        return self._output_parser.to_dict()
+    def _store_output(self, output_dict):
+        output_dict_to_hdf(
+            data_dict=output_dict,
+            hdf=self._hdf5,
+            group_name="output",
+        )
+        if len(self._exclude_groups_hdf) > 0 or len(self._exclude_nodes_hdf) > 0:
+            self.project_hdf5.rewrite_hdf5()
```

Comment:
Can't we just put this directly in collect_output? Is this being used somewhere else?

Reply:
Yes, the idea is to move the writing of the input, the execution of the executable and the parsing of the output into a separate function which can then be submitted to an `pympipool` executor. This is implemented in https://github.com/pyiron/pyiron_atomistics/pull/1454 which is based on the changes in this pull request.

Reply:
if they are the same function (as I am seeing here), what is the advantage of creating a new hidden method vs just calling collect_output?

Reply:
The current `collect_output()` function consists of two parts, the parsing of the output files and the storing of the output inside the HDF5 file. These two parts are separated in future into the `parse_vasp_output()` function which could even be called as a standalone output parser and the `job._store_output() function. To remain compatible to the previous interface the `collect_output()` function calls these two functions. In future, the parsing of the output might happen on one compute node while the storing in the HDF5 file happens on the central node of the allocation, to prevent multiple nodes accessing the same HDF5 file.